test/test_irods_http_api.py

```diff
@@ -3339,6 +3339,238 @@ def test_rodsadmin_can_invoke_truncate_on_other_users_data_objects_using_the_adm
+
+        finally:
+            # Remove the data object.
+            r = requests.post(self.url_endpoint, headers=rodsuser_headers, data={
+                'op': 'remove',
+                'lpath': data_object,
+                'catalog-only': 0,
+                'no-trash': 1
+            })
+            self.logger.debug(r.content)
+
+            # Remove the resource.
+            r = requests.post(f'{self.url_base}/resources', headers=rodsadmin_headers, data={
+                'op': 'remove',
+                'name': resc_name
+            })
+            self.logger.debug(r.content)
+
+    def test_targeting_replica_using_invalid_replica_number_or_resource_results_in_an_error(self):
+        rodsuser_headers = {'Authorization': f'Bearer {self.rodsuser_bearer_token}'}
+        data_object = f'/{self.zone_name}/home/{self.rodsuser_username}/using_invalid_replica_numbers_and_resources.txt'
+
+        try:
+            # Create a data object.
+            r = requests.post(self.url_endpoint, headers=rodsuser_headers, data={
+                'op': 'touch',
+                'lpath': data_object
+            })
+            self.logger.debug(r.content)
+            self.assertEqual(r.status_code, 200)
+            self.assertEqual(r.json()['irods_response']['status_code'], 0)
+
+            with self.subTest('Writing to replica using non-integer string as replica number'):
+                r = requests.post(self.url_endpoint, headers=rodsuser_headers, data={
+                    'op': 'write',
+                    'lpath': data_object,
+                    'replica-number': 'invalid',
+                    'bytes': 'bad replica number'
+                })
+                self.logger.debug(r.content)
+                self.assertEqual(r.status_code, 400)
+
+            with self.subTest('Writing to replica using incorrect replica number'):
+                r = requests.post(self.url_endpoint, headers=rodsuser_headers, data={
+                    'op': 'write',
+                    'lpath': data_object,
+                    'replica-number': 100,
+                    'bytes': 'bad replica number'
+                })
+                self.logger.debug(r.content)
+                self.assertEqual(r.status_code, 200)
+                self.assertEqual(r.json()['irods_response']['status_code'], irods_error_codes.INVALID_HANDLE)
+
+            with self.subTest('Writing to replica using invalid resource'):
+                r = requests.post(self.url_endpoint, headers=rodsuser_headers, data={
+                    'op': 'write',
+                    'lpath': data_object,
+                    'resource': 'doesnotexist',
+                    'bytes': 'bad resource'
+                })
+                self.logger.debug(r.content)
+                self.assertEqual(r.status_code, 200)
+                self.assertEqual(r.json()['irods_response']['status_code'], irods_error_codes.INVALID_HANDLE)
```

Comment:
Cover different methods of the `write` operation - i.e. header-based form and multipart.